packages/devtools_app/lib/src/screens/inspector/inspector_data_models.dart

```diff
@@ -439,62 +454,62 @@ class FlexLayoutProperties extends LayoutProperties {
   }
 
   Axis get crossAxisDirection {
-    return direction == Axis.horizontal ? Axis.vertical : Axis.horizontal;
+    return direction;
   }
 
-  double get mainAxisDimension => dimension(direction);
+  double? get mainAxisDimension => dimension(direction);
 
-  double get crossAxisDimension => dimension(crossAxisDirection);
+  double? get crossAxisDimension => dimension(crossAxisDirection);
 
   @override
   bool get isOverflowWidth {
     if (direction == Axis.horizontal) {
-      return width + overflowEpsilon < sum(childrenWidths);
+      return width! + overflowEpsilon < sum(childrenWidths!.cast<double>());
     }
-    return width + overflowEpsilon < max(childrenWidths);
+    return width! + overflowEpsilon < max(childrenWidths!.cast<double>());
   }
 
   @override
   bool get isOverflowHeight {
     if (direction == Axis.vertical) {
-      return height + overflowEpsilon < sum(childrenHeights);
+      return height! + overflowEpsilon < sum(childrenHeights!.cast<double>());
     }
-    return height + overflowEpsilon < max(childrenHeights);
+    return height! + overflowEpsilon < max(childrenHeights!.cast<double>());
   }
 
   bool get startIsTopLeft {
-    assert(direction != null);
     switch (direction) {
       case Axis.horizontal:
         switch (textDirection) {
           case TextDirection.ltr:
+          case null:
```

Comment:
this shouldn't be necessary since direction is non-nullable

Reply:
direction is not nullable, but textDirection is nullable.